tactics/tacticals.ml

```diff
@@ -434,9 +434,13 @@ module New = struct
     tclIFTHENELSE t1 (tclTRY t2) t2
 
   (* Try the first tactic that does not fail in a list of tactics *)
-  let rec tclFIRST = function
-    | [] -> tclZEROMSG (str"No applicable tactic.")
-    |  t::rest -> tclORELSE0 t (tclFIRST rest)
+  let rec tclFIRST ?(msg = str "No applicable tactic.") = function
+    | [] -> tclZEROMSG msg
+    | t::rest -> tclORELSE0 t (tclFIRST ~msg rest)
+
+  let rec tclMULTIFIRST ?(msg = str "No applicable tactic.") = function
```

Comment:
Indeed there is! I was not aware of that.
Can I ask why some of these primitives are in `tactics.ml` and some are in `tacticals.ml`? To me, it does not make much sense.
There is a slight difference in the code: `tclANY` uses `tclORD` instead of `tclOR`. It makes the code harder to read but does it bring performance improvement?